impl/command_line/arguments.cxx

```diff
@@ -161,14 +162,44 @@ namespace substrate::commandLine
 		}, item);
 	}
 
+	[[nodiscard]] static size_t checkExclusivity(const optionsVisited_t &options)
+	{
+		std::set<option_t> exclusiveOptions{};
+		// Loop through all the visited options
+		for (const auto &option : options)
+		{
+			// Dispatch on the option type
+			std::visit(match_t
+			{
+				[&](const option_t &value)
+				{
+					if (value.isExclusive())
+						exclusiveOptions.insert(value);
+				},
+				[](const optionSet_t &) { },
+			}, option);
+		}
+		// Now we've collected all the options that are exclusive, display diagnostics
+		// if there is more than one in the set and then return how many there are
+		if (exclusiveOptions.size() > 1)
+		{
+			console.error("Multiple mutually exclusive options given together on command line, only one allowed."sv);
+			console.error("Mutually exclusive options given are:"sv);
+			for (const auto &option : exclusiveOptions)
+				console.error("    "sv, option.displayName());
+		}
+		return exclusiveOptions.size();
+	}
+
 	// NOLINTNEXTLINE(readability-convert-member-functions-to-static)
-	bool arguments_t::parseFrom(tokeniser_t &lexer, const options_t &options)
+	// NOLINTNEXTLINE(performance-unnecessary-value-param)
+	bool arguments_t::parseFrom(tokeniser_t &lexer, const options_t &options, const optionsVisited_t globalOptions)
```

Comment:
```suggestion
	bool arguments_t::parseFrom(tokeniser_t &lexer, const options_t &options, const optionsVisited_t globalOptions) noexcept
```

(unsure on this case)

Reply:
collectRequiredOptions() and parseArgument() can both throw here due to STL container operations involving allocations, so parseFrom() can unfortunately also throw.

Reply:
Throughout the code you've made the assumption that e.g. the `std::variant` cast can fail, and that should instantly cause a termination. Why isn't the case here, since a `bad_alloc` (pun not intended) is also a mostly unrecoverable failure?

Reply:
Ah, hmm.. so, the actual assumption we thought we'd made was that: the variants all always hold at least one valid alternative, so visiting all valid alternatives is always safe, so the only way a `std::variant` `std::visit` or `std::get` can fail is if no valid alternatives are held which would be a programming mistake - something that would preferably be compile-time discovered.

The difference is that allocation failure is a runtime issue so we should make that fail as gracefully as an OOM condition allows, we think.

Reply:
I'll approve this based on the comment regarding the expectation around argument parsing failure -- either a bad_alloc or a bad_variant (the latter of which is always considered a hard failure inside Substrate).

Reply:
The more we think on it the more we come to the conclusion the only two kinds of exception that can emanate from this function are: `std::variant` conversion failures, which we're letting get thrown so they're trivial to debug as they're programming errors; and `std::bad_alloc` which is extremely fatal anyway, so probably best left to propagate through top-level `parseArguments()` to cause a std::terminate().

If you can think of any others then what should probably happen is that the top-level parseArguments() should catch, display and translate them - but no exceptions should actually be able to leave the parser as-is and this will automatically unwind the stack for us as currently written.